solidity/contracts/relays/OpenRelay.sol

```diff
@@ -17,25 +17,56 @@ contract OpenRelay is IOpenRelay {
   /// @inheritdoc IOpenRelay
   uint32 public constant BASE = 10_000;
 
+  mapping(IAutomationVault _automationVault => mapping(address _job => PaymentData _paymentData)) public paymentsData;
+
+  function setExtraPayment(IAutomationVault _automationVault, address _job, PaymentData memory _paymentData) external {
+    if (_automationVault.owner() != msg.sender) revert OpenRelay_NotVaultOwner();
+
+    paymentsData[_automationVault][_job] = _paymentData;
+
+    emit ExtraPaymentSetted(_automationVault, _job, _paymentData);
+  }
+
   /// @inheritdoc IOpenRelay
   function exec(
     IAutomationVault _automationVault,
     IAutomationVault.ExecData[] calldata _execData,
     address _feeRecipient
   ) external {
     if (_execData.length == 0) revert OpenRelay_NoExecData();
+    IAutomationVault.ExecData[] memory _newExecData = new IAutomationVault.ExecData[](1);
+    IAutomationVault.FeeData[] memory _feeData = new IAutomationVault.FeeData[](_execData.length);
+    PaymentData memory _paymentData;
+
+    uint256 _initialGas;
+    uint256 _gasSpent;
+    uint256 _payment;
+
+    for (uint256 _i; _i < _execData.length;) {
+      _paymentData = paymentsData[_automationVault][_execData[_i].job];
+      _newExecData[0] = _execData[_i];
+
+      // Execute the automation vault counting the gas spent
+      _initialGas = gasleft();
+      _automationVault.exec(msg.sender, _newExecData, new IAutomationVault.FeeData[](0));
+      _gasSpent = _initialGas - gasleft();
 
-    // Execute the automation vault counting the gas spent
-    uint256 _initialGas = gasleft();
-    _automationVault.exec(msg.sender, _execData, new IAutomationVault.FeeData[](0));
-    uint256 _gasSpent = _initialGas - gasleft();
+      if (address(_paymentData.token) == _NATIVE_TOKEN) {
+        _payment = (_gasSpent + GAS_BONUS) * block.basefee * _paymentData.amountOrPercentage / BASE;
+        _feeData[_i] = IAutomationVault.FeeData(_feeRecipient, _NATIVE_TOKEN, _payment);
+      } else if (address(_paymentData.token) == address(0)) {
+        _payment = (_gasSpent + GAS_BONUS) * block.basefee * GAS_MULTIPLIER / BASE;
+        _feeData[_i] = IAutomationVault.FeeData(_feeRecipient, _NATIVE_TOKEN, _payment);
```

Comment:
this is weird, both `_NATIVE_TOKEN` and `address(0)` relate to native payment if i'm not mistaken, yet using the 1st you're calculating with `amountOrPercentage`, and with the 2nd you're using `GAS_MULTIPLIER`, why?

Reply:
In this case I thought:

1 If the owner did not define any incentive for the job, the normal payment is calculated.

2 If he wants to opt for another percentage, he puts in the mapping native token and it is calculated with the custom percentage. 

3 If he wants to pay with an extra token he defines the token